patches/server/0745-Make-EntityUnleashEvent-cancellable.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Sun, 3 Jan 2021 21:25:31 -0800
+Subject: [PATCH] Make EntityUnleashEvent cancellable
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
+index a8e5be1c941755b3e5b335d8211ca70a6c6fc32f..fc010e58255f4949fafe1df8683b18c6354dcac8 100644
+--- a/src/main/java/net/minecraft/world/entity/Mob.java
++++ b/src/main/java/net/minecraft/world/entity/Mob.java
+@@ -1448,7 +1448,7 @@ public abstract class Mob extends LivingEntity {
+         if (flag1 && this.isLeashed()) {
+             // Paper start - drop leash variable
+             EntityUnleashEvent event = new EntityUnleashEvent(this.getBukkitEntity(), UnleashReason.UNKNOWN, true);
+-            this.level.getCraftServer().getPluginManager().callEvent(event); // CraftBukkit
++            if (!event.callEvent()) { return flag1; }
+             this.dropLeash(true, event.isDropLeash());
+             // Paper end
+         }
+@@ -1624,7 +1624,7 @@ public abstract class Mob extends LivingEntity {
+         super.removeAfterChangingDimensions();
+         // Paper start - drop leash variable
+         EntityUnleashEvent event = new EntityUnleashEvent(this.getBukkitEntity(), UnleashReason.UNKNOWN, false);
+-        this.level.getCraftServer().getPluginManager().callEvent(event); // CraftBukkit
++        if (!event.callEvent()) { return; }
```

Comment:
Returning here looks *really* scary

Reply:
Yeah, it certainly does, but like if the player just follows through right after, everything works out. Is that leaving too much "freedom" up to plugin devs that might mess things up?

Reply:
Mm yeah definitely add a return there. The rest is good to go, not much we can do to make cancelling consistent without breakage

Reply:
...Do you mean definitely **don't** add a return here? I have a feeling you meant the opposite of what you said.

Reply:
Yes, whoopsie